### PR TITLE
Fix language changing issues

### LINKED ISSUE #270: Language does not change on a few devices
Language does not change on a physical device Samsung S10 plus Android 11.

Settings appear to be saved but app still shows in English. App does not even restart after language change for a few languages.

Works well on Android 10 and 11 VMs.

Tested reinstalling the app and also cleared all data. 

@spuday90 any idea what could be going wrong?

### PR DESCRIPTION
**Pull request by hegocre**
Fix for locale changing issues, possible fix #270, tested on Android Emulator API 21 (previously not working, now it does), as well as API 29 and 30 on physical devices, both working.

When creating locales, region code must be a 2 alpha character string as explained [here](https://developer.android.com/reference/java/util/Locale), I imagine the r added on resource folder is an abbreviation for region and should not be used when creating Locales in code.

Also, a bug on AndroidX compat library causes the locale to be overridden to default on API versions 21 to 25 because of DayNight compatibility, so applyOverrideConfiguration must be called with the new context created.

### CODE REVIEW

**Review by adeekshith — APPROVED** [review]
Thanks a lot @hegocre for this as well as your Spanish and Catalan translations 😃

Calling `applyOverrideConfiguration` itself fixed the issue on the device I was testing without changing the language code.

Do not see this affecting existing users too but will wait for @spuday90 to review this before merging.

Thank you!!

**Comment by hegocre** [comment]
@adeekshith You are welcome!

I also noticed that some devices neee none, some need one and others need both patches to work correctly.

The problem I see here is that existing users will need to set their language again. Maybe changing PreferencesManager.java to remove the r before creating the locale would be lesw intrusive, but would require you to keep adding the r to new locales.

If you prefer it that way I can change this on my pull request before merging.

Cheers!

**Comment by adeekshith** [comment]
> The problem I see here is that existing users will need to set their language again.

Oh I was thinking we were saving the key (`lang_code_ca_rES`) and not the value (`ca-ES`). I am wrong then. Thanks for catching.

> Maybe changing PreferencesManager.java to remove the r before creating the locale would be lesw intrusive, but would require you to keep adding the r to new locales.

That should do but how about this solution:
1. Change [`PreferencesManager::getSelectedLanguageStr()`](https://github.com/adeekshith/watomatic/blob/d7178ed48d752c9f99512f1eb6e0c320f497f051/app/src/main/java/com/parishod/watomatic/model/preferences/PreferencesManager.java#L183-L185) method to remove `r` before returning (`ca-rES` to `ca-ES`).
2. Or even better, we can write code to migrate existing `r` version to the non-r version in the shared preferences [`PreferencesManager::init()`](https://github.com/adeekshith/watomatic/blob/d7178ed48d752c9f99512f1eb6e0c320f497f051/app/src/main/java/com/parishod/watomatic/model/preferences/PreferencesManager.java#L56-L78) method (init() method executed when the PreferencesManager object is created). Advantage with this method is, we can remove the migration code after a few months without affecting anyone.

**Comment by adeekshith** [comment]
Awesome! Your fix (186f302) looks good @hegocre 
Will test it in a while and merge 😃

Thank you!!

**Comment by hegocre** [comment]
You're welcome!

Amazing work with this app!

**Review by spuday90 — APPROVED** [review]

**Review by adeekshith — APPROVED** [review]

**Pull request merged**
